control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -336,10 +290,195 @@ func TestReconcileCreateEndpointWithNamespaces(t *testing.T) {
 				var checks api.HealthChecks
 				filter := fmt.Sprintf("CheckID == `%s`", expectedCheck.CheckID)
 				checks, _, err := consulClient.Health().Checks(expectedCheck.ServiceName, &api.QueryOptions{Filter: filter})
-				if expectedCheck.ServiceName == "mesh-gateway" {
-					checks, _, err = consulClient.Health().Checks("mesh-gateway", &api.QueryOptions{Namespace: "default"})
-					require.NoError(t, err)
+				require.NoError(t, err)
+				require.Equal(t, len(checks), 1)
+				var ignoredFields = []string{"Node", "Definition", "Partition", "CreateIndex", "ModifyIndex", "ServiceTags"}
+				require.True(t, cmp.Equal(checks[0], expectedCheck, cmpopts.IgnoreFields(api.HealthCheck{}, ignoredFields...)))
+			}
+		})
+	}
+}
+
+// TestReconcileCreateGatewayWithNamespaces verifies that gateways created using
+// the Endpoints Controller with Consul namespaces are correct.
+func TestReconcileCreateGatewayWithNamespaces(t *testing.T) {
+	t.Parallel()
+	cases := map[string]struct {
+		ConsulNS string
+	}{
+		"default Consul namespace": {
+			ConsulNS: "default",
+		},
+		"other Consul namespace": {
+			ConsulNS: "other",
+		},
+	}
+	for name, testCase := range cases {
+		setup := struct {
+			k8sObjects                 func() []runtime.Object
+			expectedConsulSvcInstances []*api.CatalogService
+			expectedProxySvcInstances  []*api.CatalogService
+			expectedHealthChecks       []*api.HealthCheck
+		}{
+			k8sObjects: func() []runtime.Object {
+				meshGateway := createGatewayWithNamespace("mesh-gateway", "default", "3.3.3.3", map[string]string{
```

Comment:
I think it makes sense to leave it so that it is similar to the existing `createPodWithNamespace` function and just in case we need to modify the namespace in future tests. I'll give you veto power on that decision. Just let me know.